ops/log.py

```diff
@@ -53,6 +53,7 @@ def setup_root_logging(
         debug: if True, write logs to stderr as well as to juju-log.
         exc_stderr: if True, write uncaught exceptions to stderr as well as to juju-log.
     """
+    logging.captureWarnings(True)
```

Comment:
I'm still of the opinion that we should have a custom formatter. There's so much that goes into the debug-log - I think we need to take care to try not to make things worse. I don't see the value in repeating the part of the code that triggered the warning - either the charm is open-source, in which case the code can be found fairly easily<sup>*</sup> via the file and line number, or it's not (is this ever the case?) and we shouldn't be leaking the code.

In either situation, my expectation would be that the charm should provide in the message instructions for the Juju admin (use a different action name/param, upgrade a related charm, etc). It seems doubtful that having the line of code will help with that. (Even the file and line don't seem helpful, except perhaps if you're communicating with the charm author and they haven't made it easy to describe).

As discussed in the other PR, I think the full filename is more than what's required, but also that trying to properly trim off the parts we don't care about is maybe too much to do right now.

<sup>*</sup> To be fair, it's harder than it should be to find which version of the charm is running, much of the time (charms could solve this by adding the VERSION file, but most don't, and there are documentation issues there). So file+line number is ideally combined with revision, which then gets used to (hopefully) look up a git tag.

Reply:
I'm ambivalent on custom formatter.

Standard format means lesser mental load on library user; custom format is more concise, thus also lesser mental load.

Re "leaking" source code lines. If we went that way, we'd have to whiten the traceback in `ops/log.py: except_hook` too. Somehow I don't see us doing that at the moment.